bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,172 @@
+                                                  unescaped_search_in_between)
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.HiddenResult import HiddenResult
+
+
+class AnnotationBear(LocalBear):
+
+    def run(self, filename, file: str, language: str, language_family: str):
+        """
+        Finds out all the positions of comments and strings in a file.
+        The Bear searches for valid comments and strings
+        and yields them as HiddenResults with the appropriate ranges.
+        Invalid comments or strings are:
+        1. Comments within comments,
+        2. Comments within strings and
+        3. Strings within comments.
+        """
+        file = ''.join(file)
+        annot_dict = {}
+        annot_dict['comment'] = {}
+        annot_dict['string'] = {}
+        annot_dict['string']['multiline'] = {}
+        annot_dict['string']['singleline'] = {}
+        annot_dict['comment']["singleline"] = {}
+        annot_dict['comment']["multiline"] = {}
+        # Tuple of tuples containing start and end of annotations
+        match_pos = ()
+        lang_dict = LanguageDefinition(language, language_family)
+
+        annot_dict['comment']["singleline"].update(
+                    lang_dict["comment_delimiter"])
+
+        annot_dict['comment']["multiline"].update(
+                    lang_dict["multiline_comment_delimiters"])
+
+        annot_dict['string']['singleline'].update(
+                                            lang_dict["string_delimiters"])
+        annot_dict['string']['multiline'].update(
+                                       lang_dict["multiline_string_delimiters"])
+        multi_str = single_str = multi_comm = ()
+        search_dict = annot_dict['string']['multiline']
+        multi_str = find_string_comment(file, search_dict, True)
+        search_dict = annot_dict['string']['singleline']
+        single_str = find_string_comment(file, search_dict, True)
+        search_dict = annot_dict['comment']["multiline"]
+        multi_comm = find_string_comment(file, search_dict, False)
+
+        match_pos += single_str + multi_str + multi_comm
+        # Seperate since single-line comments don't have a definite start and
+        # end
+        search_dict = annot_dict['comment']["singleline"]
+        single_comm = find_singleline_comments(file, search_dict, match_pos)
+        match_pos += single_comm
+        match_pos = remove_nested(match_pos)
+        m_comm = s_comm = m_str = s_str = []
+        if match_pos:
+            m_comm = list(get_range(file, filename, match_pos, multi_comm))
+            s_comm = list(get_range(file, filename, match_pos, single_comm))
+            m_str = list(get_range(file, filename, match_pos, multi_str))
+            s_str = list(get_range(file, filename, match_pos, single_str))
+        yield HiddenResult(self, m_comm)
```

Comment:
This can reduce repetition:

```
if match_pos:
    for type in [multi_comm, single_comm, multi_str, single_str]:
        yield HiddenResult(self, list(get_range(file, filename, match_pos, type))
```
PS: I dont really know what these are yet ... so, you should use a better name that `type`.  Just used that as a dummy